app/src/main/java/org/mozilla/reference/browser/freshtab/FreshTabFragment.kt

```diff
@@ -0,0 +1,85 @@
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import com.cliqz.browser.news.ui.NewsInteractor
+import kotlinx.android.synthetic.main.fragment_fresh_tab.*
+import kotlinx.android.synthetic.main.news_layout.*
+import kotlinx.android.synthetic.main.top_sites_layout.*
+import mozilla.components.support.base.feature.UserInteractionHandler
+import mozilla.components.support.base.feature.ViewBoundFeatureWrapper
+import org.mozilla.reference.browser.BrowserActivity
+import org.mozilla.reference.browser.R
+import org.mozilla.reference.browser.ext.isFreshTab
+import org.mozilla.reference.browser.ext.requireComponents
+
+class FreshTabFragment : Fragment(), UserInteractionHandler {
+
+    private val freshTabIntegration = ViewBoundFeatureWrapper<FreshTabIntegration>()
+
+    lateinit var freshTabInteractor: FreshTabInteractor
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_fresh_tab, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val freshTabController = DefaultFreshTabController(
+            activity as BrowserActivity,
+            requireComponents.core.sessionManager,
+            requireComponents.useCases.tabsUseCases,
+            findNavController(),
+            lifecycleScope
+        )
+
+        freshTabInteractor = FreshTabInteractor(freshTabController)
+
+        freshTabIntegration.set(
+            feature = FreshTabIntegration(
+                requireContext(),
+                requireComponents.core.sessionManager,
+                fresh_tab_toolbar)
+                .addToolbarFeature(freshTabInteractor, lifecycleScope)
+                .addNewsFeature(
+                    newsView,
+                    lifecycleScope,
+                    freshTabInteractor as NewsInteractor,
+                    requireComponents.useCases.getNewsUseCase,
+                    requireComponents.core.icons)
+                .addTopSitesFeature(
+                    topSitesView,
+                    requireComponents.useCases.historyUseCases,
+                    requireComponents.core.icons),
+            owner = this,
+            view = view
+        )
+    }
+
+    override fun onBackPressed(): Boolean {
+        return removeSessionIfNeeded()
+    }
+
+    private fun removeSessionIfNeeded(): Boolean {
+        val sessionManager = requireComponents.core.sessionManager
+        if (sessionManager.selectedSession != null &&
+                !sessionManager.selectedSession!!.isFreshTab()) {
+            sessionManager.remove()
+        }
+        return false
+    }
+}
```

Comment:
We have similar methods in BaseBrowserFragment and BrowserActivity

Reply:
Do you mean why am I removing the session?
This is because the user navigated to fresh tab while still having a session which is not 'about:blank' and so then the user would want to be on the fresh tab when reopening the app again.